engine/src/main/java/org/terasology/world/block/family/EdgeBlockFamilyFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.block.family;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.math.Edge;
+import org.terasology.math.Rotation;
+import org.terasology.math.Side;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
+import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
+
+import java.util.EnumMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+
+@RegisterBlockFamilyFactory("edge")
+public class EdgeBlockFamilyFactory implements BlockFamilyFactory {
+    private static Logger logger = LoggerFactory.getLogger(EdgeBlockFamilyFactory.class);
+    private static final ImmutableSet<String> BLOCK_NAMES = ImmutableSet.of("front", "left", "right", "back", "top", "bottom");
+    private static final ImmutableList<MultiSection> MULTI_SECTIONS = ImmutableList.of(
+            new MultiSection("all", "front", "left", "right", "back", "top", "bottom"),
+            new MultiSection("topBottom", "top", "bottom"),
+            new MultiSection("sides", "front", "left", "right", "back"));
+
+    @Override
+    public BlockFamily createBlockFamily(BlockFamilyDefinition definition, BlockBuilderHelper blockBuilder) {
+        throw new IllegalStateException("A shape must be provided when creating a family for a edge block family definition");
+    }
+
+    @Override
+    public BlockFamily createBlockFamily(BlockFamilyDefinition definition, BlockShape shape, BlockBuilderHelper blockBuilder) {
+        if (!definition.isFreeform()) {
+            throw new IllegalStateException("A shape cannot be provided when creating a family for a non-freeform block family definition");
+        }
+        BlockUri uri = new BlockUri(definition.getUrn(), shape.getUrn());
+
+        Edge archetypeEdge = Edge.of(shape.getBlockShapePlacement().getArchetype());
+        if( archetypeEdge != Edge.BottomBack ) {
+            Rotation rotToBottomBlock = archetypeEdge.getRotationFromBottomBack();
+            //Fixme: invert and create base for following rotations...
```

Comment:
Maybe better a TODO. I will not implement this, becaue it is somewhat complicated. A Rotation cannot give its inverse. This functionality will be used if someone builds a shape with edge placement, which is not oriented that BOTTOM_BACK is a proper base orientation. As far as I see all existing slope shapes doesn't need this to be implemented.